Merge Two Sorted Lists

```diff
@@ -0,0 +1,53 @@
+// Author: Emre Çelik
+// Question Link: https://leetcode.com/problems/merge-two-sorted-lists/
+
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     ListNode *next;
+ *     ListNode() : val(0), next(nullptr) {}
+ *     ListNode(int x) : val(x), next(nullptr) {}
+ *     ListNode(int x, ListNode *next) : val(x), next(next) {}
+ * };
+ */
+
+
+// In this problem we have 2 sorted list and we need to merge 2 of them.
+
+                // 1->2->4
+                // 1->3->4
+            // 1->1->2->3->4->4
+
+class Solution {
+public:
+    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
+        
+        // Return list that is not empty when one of our list is empty.
+        if(list1 == NULL){
+            return list2;
+        }
+        if(list2 == NULL){
+            return list1;
+        }
+            
+        //Define our answer list
+        ListNode* ans;
+        
+        //If first list has bigger value than second list call ans->next recursively with list1->next.
+        if(list1 -> val <= list2 -> val){
+            ans = list1;
+            ans -> next = mergeTwoLists(list1->next,list2);
+        }
+        // Else call ans->next recursively with list2->next.
+        else{
+            ans = list2;
+            ans -> next = mergeTwoLists(list1,list2->next);
+        }
+        //Return our answer list.
+        return ans;
+    }
+};
+
+
+
```

Comment:
a space can be added after each comma in lines 40&45 as well as the 'else' in line 43
unnecessary new lines at the end of file